### PR TITLE
refs[TASK][31133]: Create SQLiteOpenHelper,LocalDataSource

### PR DESCRIPTION
**Pull request by KhacTung-Shjn**
## Related Tickets
- [Task#31133: Create SQLiteOpenHelper and create Local Data source](https://edu-redmine.sun-asterisk.vn/issues/31133)
## WHAT
- Create SQLiteOpenHelper and Create Local Data Source
## HOW
- Create SQLiteOpenHelper and Create Local Data Source
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/premierleague/data/model/NotificationMatch.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import android.content.ContentValues
+
+data class NotificationMatch(
+    val matchId: String,
+    val matchDate: String,
+    val matchTime: String,
+    val matchHomeTeamId: String,
+    val matchHomeTeamName: String,
+    val matchAwayTeamId: String,
+    val matchAwayTeamName: String
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/premierleague/data/model/NotificationMatch.kt]:
```suggestion
data class MatchNotification(
    val id: String,
    val date: String,
    val time: String,
    val homeTeamId: String,
    val homeTeamName: String,
    val awayTeamId: String,
    val awayTeamName: String
```

```kotlin
    val homeTeamId: String,
    val homeTeamName: String,
    val awayTeamId: String,
    val awayTeamName: String
```
cái này nên tạo ra class Team và có thuộc tính check xem là `home` hay `away` sẽ tốt hơn

File: app/src/main/java/com/sun/premierleague/data/model/NotificationMatch.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+) {
+    fun getContentValues() = ContentValues().apply {
+        put(MatchItem.MATCH_ID, matchId)
+        put(MatchItem.MATCH_DATE, matchDate)
+        put(MatchItem.MATCH_TIME, matchTime)
+        put(MatchItem.MATCH_HOMETEAM_ID, matchHomeTeamId)
+        put(MatchItem.MATCH_HOMETEAM_NAME, matchHomeTeamName)
+        put(MatchItem.MATCH_AWAYTEAM_ID, matchAwayTeamId)
+        put(MatchItem.MATCH_AWAYTEAM_NAME, matchAwayTeamName)
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/premierleague/data/model/NotificationMatch.kt]: đống này lấy từ companion object của class khác vào à em?

File: app/src/main/java/com/sun/premierleague/data/model/NotificationMatch.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    val matchDate: String,
+    val matchTime: String,
+    val matchHomeTeamId: String,
+    val matchHomeTeamName: String,
+    val matchAwayTeamId: String,
+    val matchAwayTeamName: String
+) {
+    fun getContentValues() = ContentValues().apply {
+        put(MatchItem.MATCH_ID, matchId)
+        put(MatchItem.MATCH_DATE, matchDate)
```
Comment (huytq-2422) [inline app/src/main/java/com/sun/premierleague/data/model/NotificationMatch.kt]:
```suggestion
        put(MATCH_ID, matchId)
        put(MATCH_DATE, matchDate)
```

File: app/src/main/java/com/sun/premierleague/data/source/local/dao/NotificationMatchDAOImpl.kt [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+                MatchNotification(
+                    matchID,
+                    matchDate,
+                    matchTime,
+                    matchHomeID,
+                    matchHomeName,
+                    matchAwayID,
+                    matchAwayName
+                )
+            )
```
Comment (huytq-2422) [inline app/src/main/java/com/sun/premierleague/data/source/local/dao/NotificationMatchDAOImpl.kt]: Tạo một constructor truyền vào cursor cho MatchNotification nhé

File: app/src/main/java/com/sun/premierleague/data/source/local/database/NotificationMatchDatabase.kt [hunk 5]
```diff
@@ -0,0 +1,49 @@
+package com.sun.premierleague.data.source.local.database
+
+import android.content.Context
+import android.database.sqlite.SQLiteDatabase
+import android.database.sqlite.SQLiteOpenHelper
+import com.sun.premierleague.data.model.MatchNotification
+import com.sun.premierleague.data.model.NotificationMatch
+
+class NotificationMatchDatabase(context: Context, dbName: String, version: Int) :
```
Comment (huytq-2422) [inline app/src/main/java/com/sun/premierleague/data/source/local/database/NotificationMatchDatabase.kt]:
```suggestion
class AppDatabase private constructor(context: Context, dbName: String, version: Int) :
```

File: app/src/main/java/com/sun/premierleague/data/model/NotificationMatch.kt (hunk 1 above)
Comment (thonc-2135) [inline app/src/main/java/com/sun/premierleague/data/model/NotificationMatch.kt]: sửa như đạt ns đi, k ai đi đặt thêm match như v đâu, thừa. tại sao thì đọc cuốn clean code

File: app/src/main/java/com/sun/premierleague/data/source/local/dao/NotificationMatchDAOImpl.kt [hunk 6]
```diff
@@ -0,0 +1,102 @@
+package com.sun.premierleague.data.source.local.dao
+
+import com.sun.premierleague.data.model.MatchNotification
+import com.sun.premierleague.data.model.NotificationMatch
+import com.sun.premierleague.data.source.local.database.NotificationMatchDatabase
+
+class NotificationMatchDAOImpl(notificationMatchDatabase: NotificationMatchDatabase) :
+    NotificationMatchDAO {
```
Comment (huytq-2422) [inline app/src/main/java/com/sun/premierleague/data/source/local/dao/NotificationMatchDAOImpl.kt]: singleton để private constructor ông nhé

**Pull request merged**